src/validators/mod.rs

```diff
@@ -31,26 +31,43 @@ mod union;
 #[pyclass]
 #[derive(Debug, Clone)]
 pub struct SchemaValidator {
-    validator: CombinedValidator,
+    validator: ValidateEnum,
     slots: Vec<CombinedValidator>,
+    schema: Py<PyAny>,
+}
+
+fn build_schema_validator(py: Python, schema: &PyAny) -> PyResult<SchemaValidator> {
+    let mut slots_builder = SlotsBuilder::new();
+    let validator = match build_validator(schema, None, &mut slots_builder) {
+        Ok((v, _)) => v,
+        Err(err) => {
+            return Err(match err.is_instance_of::<SchemaError>(py) {
+                true => err,
+                false => SchemaError::new_err(format!("Schema build error:\n  {}", err)),
+            });
+        }
+    };
+    let slots = slots_builder.into_slots()?;
+    Ok(Self { validator, slots })
+}
+
+#[pyfunction]
+fn build_schema_validator_py(schema: &PyAny) -> PyResult<SchemaValidator> {
+    Python::with_gil(|py| build_schema_validator(py, schema))
 }
 
 #[pymethods]
 impl SchemaValidator {
     #[new]
+    pub fn py_new(py: Python, schema: &PyAny) -> PyResult<SchemaValidator> {
+        build_schema_validator(py, schema)
+    }
+
+    fn __reduce__<'py>(&self, py: Python<'py>) -> &'py PyTuple {
+        PyTuple::new(py, vec![self.schema.as_ref(py)]) // TODO: return a ref to build_schema_validator_py?
```

Comment:
We'd have to return a reference to the Python version of `build_schema_validator_py` here, I'm not sure how to do that